src/main.rs

```diff
@@ -126,20 +128,36 @@ struct CliArgs {
     reserve_size: usize,
 }
 
+#[derive(Clone, Debug)]
+enum TrustResource {
+    File(PathBuf),
+    Url(String),
+}
+
+fn parse_resource_string(s: &str) -> Result<TrustResource> {
+    if s.starts_with("http://") || s.starts_with("https://") {
+        Ok(TrustResource::Url(s.to_string()))
```

Comment:
Maybe we should bring in the `url` crate here and have it do the parsing? A couple of benefits would be:
1. You get full URL verification instead of just the scheme
2. You support all the schemes, including FTP, file, etc... Not only that, but you could build up a `file://` URL to test the URL stuff without having the tests reach out a different computer. 
3. Clap's value_parser handles the URL parsing errors instead of c2patool logic. 

```suggestion
#[derive(Clone, Debug)]
enum TrustResource {
    File(PathBuf),
    Url(url::Url),
}

fn parse_resource_string(s: &str) -> Result<TrustResource> {
    if let Ok(url) = s.parse::<url::Url>() {
        Ok(TrustResource::Url(url))
```

Reply:
@dyro Updated to use `url`.